src/Auth.php

```diff
@@ -588,21 +614,31 @@ public static function validate_token( $token = null, $refresh = false ) {
 			}
 
 			/**
-			 * If there is a user_secret in the token (refresh tokens) make sure it matches what
+			 * If there is a user_secret in the token make sure it is not revoked and still valid.
 			 */
 			if ( isset( $token->data->user->user_secret ) ) {
 
 				if ( Auth::is_jwt_secret_revoked( $token->data->user->id ) ) {
-					throw new \Exception( __( 'The User Secret does not match or has been revoked for this user', 'wp-graphql-jwt-authentication' ) );
+					throw new \Exception( __( 'The User Secret has been revoked for this user', 'wp-graphql-jwt-authentication' ) );
 				}
+
+				$token_user_secret = $token->data->user->user_secret;
+				$user_secret       = get_user_meta( $token->data->user->id, 'graphql_jwt_auth_secret', true );
+
+				if ( $token_user_secret !== $user_secret ) {
+					throw new \Exception( __( 'The User Secret does not match for this user', 'wp-graphql-jwt-authentication' ) );
+				}
+			} else if (self::$auth_token_is_stateful || self::$is_refresh_token) {
+				throw new \Exception( __( 'The User Secret is missing in the token.', 'wp-graphql-jwt-authentication' ) );
```

Comment:
Same comment as above. If the secret exists for both auth tokens and refresh tokens, the tokens are the same 🤔